src/python_testing/TC_OpstateCommon.py

```diff
@@ -0,0 +1,898 @@
+                                                cmd=commands.Start(),
+                                                expected_response=cluster.Enums.ErrorStateEnum.kNoError)
+
+        # STEP 5: TH reads from the DUT the OperationalState attribute
+        self.step(5)
+        if self.pics_guard(self.check_pics("%s.S.A0004" % self.test_info.pics_code)):
+            await self.read_and_expect_value(endpoint=endpoint,
+                                             attribute=attributes.OperationalState,
+                                             expected_value=cluster.Enums.OperationalStateEnum.kRunning)
+
+        # STEP 6: TH reads from the DUT the OperationalError attribute
+        self.step(6)
+        if self.pics_guard(self.check_pics("%s.S.A0005" % self.test_info.pics_code)):
+            await self.read_and_expect_property_value(endpoint=endpoint,
+                                                      attribute=attributes.OperationalError,
+                                                      attr_property="errorStateID",
+                                                      expected_value=cluster.Enums.ErrorStateEnum.kNoError)
+
+        # STEP 7: TH reads from the DUT the CountdownTime attribute
+        self.step(7)
+        if self.pics_guard(self.check_pics("%s.S.A0002" % self.test_info.pics_code)):
+            initial_countdown_time = await self.read_expect_success(endpoint=endpoint,
+                                                                    attribute=attributes.CountdownTime)
+            if initial_countdown_time is not NullValue:
+                asserts.assert_true(0 <= initial_countdown_time <= 259200,
+                                    "CountdownTime(%s) must be between 0 and 259200" % initial_countdown_time)
+
+        # STEP 8: TH reads from the DUT the PhaseList attribute
+        self.step(8)
+        if self.pics_guard(self.check_pics("%s.S.A0000" % self.test_info.pics_code)):
+            phase_list = await self.read_expect_success(endpoint=endpoint,
+                                                        attribute=attributes.PhaseList)
+            phase_list_len = 0
+            if phase_list is not NullValue:
+                phase_list_len = len(phase_list)
+                asserts.assert_less_equal(phase_list_len, 32,
+                                          "PhaseList length(%d) must be less than 32!" % phase_list_len)
+
+        # STEP 9: TH reads from the DUT the CurrentPhase attribute
+        self.step(9)
+        if self.pics_guard(self.check_pics("%s.S.A0001" % self.test_info.pics_code) and phase_list_len):
+            current_phase = await self.read_expect_success(endpoint=endpoint,
+                                                           attribute=attributes.CurrentPhase)
+            if (phase_list == NullValue) or (not phase_list):
+                asserts.assert_true(current_phase == NullValue, "CurrentPhase(%s) should be null" % current_phase)
```

Comment:
This check differs from the TP.

Reply:
The spec states: "This attribute SHALL be null if the PhaseList attribute is null or if the PhaseList attribute is an empty list."

I'm gonna propose the change in https://github.com/CHIP-Specifications/chip-test-plans/pull/3974

Reply:
done